gcpkms/keysource.go

```diff
@@ -243,37 +246,62 @@ func (key *MasterKey) newKMSClient() (*kms.KeyManagementClient, error) {
 	case key.credentialJSON != nil:
 		opts = append(opts, option.WithCredentialsJSON(key.credentialJSON))
 	default:
-		credentials, err := getGoogleCredentials()
-		if err != nil {
-			return nil, err
-		}
+		credentials, errCredentialsFile := getGoogleCredentials()
 		if credentials != nil {
 			opts = append(opts, option.WithCredentialsJSON(credentials))
+			break
+		}
+
+		atCredentials, errCredentialsToken := getGoogleOAuthTokenFromEnv()
+		if atCredentials != nil {
+			opts = append(opts, option.WithTokenSource(atCredentials))
```

Comment:
```suggestion
			opts = append(opts, option.WithTokenSource(atCredentials))
			break
```

Reply:
I think the `break` makes the code hard to follow and I would prefer simply keeping the `if` check below to return the error.

Reply:
I find it personally very confusing that the first `if` has a `break` and the second does not.

Also the condition for the final `if` gets rather involved if it mirrors the conditions of the two `if`s, which are no longer just `x == nil`, but now `y != nil && x == nil`.

I find the code a lot easier to understand if the conditions are not repeated all over the place in different forms, and you have to spend a lot of time thinking on what the final `if` condition actually means in relation to the previous two `if` conditions.